packages/fractor-htaccess/src/HtaccessFileProcessor.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace a9f\FractorHtaccess;
+
+use a9f\Fractor\Application\Contract\FileProcessor;
+use a9f\Fractor\Application\ValueObject\AppliedRule;
+use a9f\Fractor\Application\ValueObject\File;
+use a9f\Fractor\Caching\Detector\ChangedFilesDetector;
+use a9f\FractorHtaccess\Contract\HtaccessFractorRule;
+use Tivie\HtaccessParser\Exception\SyntaxException;
+use Tivie\HtaccessParser\HtaccessContainer;
+use Tivie\HtaccessParser\Parser;
+
+/**
+ * @implements FileProcessor<HtaccessFractorRule>
+ */
+final readonly class HtaccessFileProcessor implements FileProcessor
+{
+    /**
+     * @param iterable<HtaccessFractorRule> $rules
+     */
+    public function __construct(
+        private iterable $rules,
+        private ChangedFilesDetector $changedFilesDetector,
+        private Parser $parser
+    ) {
+    }
+
+    public function canHandle(File $file): bool
+    {
+        return $file->getFileExtension() === 'htaccess';
+    }
+
+    public function handle(File $file, iterable $appliedRules): void
+    {
+        $filePath = $file->getFilePath();
+        if (! file_exists($filePath) || ! is_readable($filePath)) {
+            return;
+        }
+
+        $fileObject = new \SplFileObject($filePath);
+
+        try {
+            /** @var HtaccessContainer $originalHtaccess */
+            $originalHtaccess = $this->parser->parse($fileObject);
+
+            // Clone doesn't work here
+            $workingHtaccess = unserialize(serialize($originalHtaccess));
```

Comment:
I have really no idea why clone doesn't work here. Do you have a better idea than to use unserialize/serialize?

Reply:
I did some research and i thought that clone would create a completely new object without any references to the old but this is somehow not the case as after the execution of the rule, the original object is also modified. The only solution is this weird unserialize/serialize combi which is quite hacky imho.